discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py

```diff
@@ -211,11 +210,33 @@ def get_primary_machine_cidr(self):
         return self._config.net_asset.machine_cidr
 
     def get_provisioning_cidr(self):
-        # In dualstack/IPv6 mode the secondary network is IPv6
-        if self.is_ipv6:
+        # In dualstack mode the primary network is IPv4
+        if self.is_ipv6 and not self.is_ipv4:
             return self._config.net_asset.provisioning_cidr6
         return self._config.net_asset.provisioning_cidr
 
+    def get_all_machine_addresses(self):
```

Comment:
To help ourselves from the future, can you add here a comment that this function gets **one** IPv4 and **one** IPv6 subnets? We know it now, but in some months we may be confused why out of multiple subnets the function returns only one

Reply:
sure, done